packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3268,6 +3272,20 @@ class _RouteEntry extends RouteTransitionRecord {
       'This route cannot be marked for pop. Either a decision has already been '
       'made or it does not require an explicit decision on how to transition out.',
     );
+    // Remove state that prevents a pop, e.g. LocalHistoryEntry[s].
+    int attempt = 0;
+    while (route.willHandlePopInternally) {
+      assert(
+        () {
+          attempt += 1;
+          return attempt < kDebugPopAttemptLimit;
+        }(),
+        'Attempted to pop $route $kDebugPopAttemptLimit times, but still failed',
+      );
+      final bool popResult = route.didPop(result);
```

Comment:
Wondering why we do this in the `markForPop` and not in the actual `handlePop` where the route is actually popped?

Also, should we be asserting somewhere that `handlePop` must actually return true if we're removing a page? This change basically disallows that a page-based route can veto its return, right?

Reply:
Wondering if this can lead to strange animations: It will basically run route-internal pop animations (e.g. to close drawers) while the regular route pop-animation runs, no?

Reply:
Would it make sense to actually add new API to the route for this to tell it: "Hey, you have to go away, if you want to do some internal pops, do them, but after that you have to go away". This would basically leave the decision if it wants to pop drawers in an animated way while the whole route is popped to the route itself.

I would image that in most case where you remove a page with an open drawer you don't really want to run the drawer close animation while the page goes away?

Reply:
> handlePop

handlePop can be called as a result of various reason, either a navigator.pop or page update. If we handle it there, I would need to add a flag and drill it down from markForPop to handlePop which is not trivial. 

> should we be asserting somewhere that handlePop must actually return true

It is possible the handlePop is a result of Navigator.pop on a page-based route. so we can't assert it return true.

> It will basically run route-internal pop animations (e.g. to close drawers) while the regular route pop-animation runs.

I don't have a counter argument to this, but if they want to do something special (if they really care) they should do it before they update the Navigator.pages.

> Would it make sense to actually add new API to the route for this to tell it: "Hey, you have to go away, if you want to do some internal pops, do them, but after that you have to go away"

I have thought about adding something like `void cleanUpInternalPopHandler()` and asserts willHandlePopInternally to return false after that, but I figure this will be another breaking change to people and will increase complexity of Route class which is already very confusing with all the canpop, popDisposition, willhandlePop...etc.

I figure the current solution is a good enough solution that will likely solve most if not all use cases. Prior to this change, the page update will just be broken. So this will only be an improvement.

If they really want to customize how route is forced to be removed, we can always add the new API later and remove this while loop didPop.